### PR TITLE
Kill processes belonging to us on install and uninstall

### PR DESCRIPTION
**Pull request by paulcbetts**
For certain types of apps, it's super easy to run the installer while the app is still running. This of course fails, but users think it is just because the installer has a bug. Enact Process Violence to stop this from occurring

### CODE REVIEW

File: src/Squirrel/NativeMethods.cs [hunk 1]
```diff
@@ -7,8 +8,30 @@
...
 {
-    static class NativeMethods
+    public static class NativeMethods
     {
+        public static int GetParentProcessId()
+        {
+            var pbi = new PROCESS_BASIC_INFORMATION();
+
+            //Get a handle to our own process
+            IntPtr hProc = OpenProcess((ProcessAccess)0x001F0FFF, false, Process.GetCurrentProcess().Id);
```
Comment (CharlieHess) [inline src/Squirrel/NativeMethods.cs]: Can you use the enum here instead of the literal?

Comment (paulcbetts) [inline src/Squirrel/NativeMethods.cs]: This is just moved code from long ago, but yeah, probably

**Comment by CharlieHess** [comment]
Could maybe beef up the tests around this scenario, although there's no way to mock them.

**Comment by ricardopolo** [comment]
@paulcbetts so this is not necesary anymore?
https://github.com/Squirrel/Squirrel.Windows/issues/358

**Comment by paulcbetts** [comment]
@ricardopolo Nope!

**Pull request merged**